aws-lc-rs/src/ec.rs

```diff
@@ -155,27 +158,100 @@ impl VerificationAlgorithm for EcdsaVerificationAlgorithm {
         msg: &[u8],
         signature: &[u8],
     ) -> Result<(), Unspecified> {
-        unsafe {
-            let ec_group = ec_group_from_nid(self.id.nid())?;
-            let ec_point = ec_point_from_bytes(&ec_group, public_key)?;
-            let ec_key = ec_key_from_public_point(&ec_group, &ec_point)?;
-
-            let ecdsa_sig = match self.sig_format {
-                EcdsaSignatureFormat::ASN1 => ecdsa_sig_from_asn1(signature),
-                EcdsaSignatureFormat::Fixed => ecdsa_sig_from_fixed(self.id, signature),
-            }?;
-            let msg_digest = digest::digest(self.digest, msg);
-            let msg_digest = msg_digest.as_ref();
-
-            if 1 != ECDSA_do_verify(msg_digest.as_ptr(), msg_digest.len(), *ecdsa_sig, *ec_key) {
-                return Err(Unspecified);
+        match self.sig_format {
+            EcdsaSignatureFormat::ASN1 => {
+                verify_asn1_signature(self.id, self.digest, public_key, msg, signature)
+            }
+            EcdsaSignatureFormat::Fixed => {
+                verify_fixed_signature(self.id, self.digest, public_key, msg, signature)
             }
-
-            Ok(())
         }
     }
 }
 
+fn verify_fixed_signature(
+    alg: &'static AlgorithmID,
+    digest: &'static digest::Algorithm,
+    public_key: &[u8],
+    msg: &[u8],
+    signature: &[u8],
+) -> Result<(), Unspecified> {
+    let mut out_bytes = MaybeUninit::<*mut u8>::uninit();
+    let mut out_bytes_len = MaybeUninit::<usize>::uninit();
+    let sig = unsafe { ecdsa_sig_from_fixed(alg, signature)? };
+    if 1 != unsafe {
+        ECDSA_SIG_to_bytes(
+            out_bytes.as_mut_ptr(),
+            out_bytes_len.as_mut_ptr(),
+            *sig.as_const(),
+        )
+    } {
+        return Err(Unspecified);
+    }
+    let out_bytes = LcPtr::new(unsafe { out_bytes.assume_init() }).map_err(|_| Unspecified)?;
```

Comment:
Here and in several other places, `map_err` shouldn't be needed.

Reply:
`LcPtr::new` returns `()` for the error type.